lib/actuators/src/actuator.h

```diff
@@ -0,0 +1,66 @@
+#include "filter.h"
+#include "drivers/actuator_drivers.h"
+#include "governors/governors.h"
+#include "converters/converters.h"
+
+class Actuator {
+  enum state_t{
+    ACTIVE,
+    ACTIVATING,
+    INACTIVE,
+    DEACTIVATING,
+    FAULT
+  };
+
+private:
+  state_t state;
+  int channel;
+  float target_value;
+  ActuatorDriver *driver;
+  Filter *filter;
+  Converter *converter;
+  Governor *governor;
+
+public:
+  //Constructor and destructor
+  Actuator(state_t s, int ch, ActuatorDriver *ad, Filter *f, Converter *c, Governor *g){
+    state = s;
+    channel = ch;
+    driver = ad;
+    filter = f;
+    converter = c;
+    governor = g;
+  }
+
+  ~Actuator(){};
+
+  //Getters and setters
+  state_t getState(){
+    return state;
+  }
+
+  void setState(state_t newState){
+    state = newState;
+  }
+
+  void set(float value){
+    if(governor->checkValue(value)){ //Validate value
```

Comment:
Probably should nomenclature change governor to validator.  A governor takes an input and limits the input to a specific range if the input is out of range (input 10,0000 rpm, max rpm is 3,000, output is 3,000).  A validator just takes and input and returns a boolean yes/no